### PR TITLE
fix(compareContents): plain type matcher not working

### PR DESCRIPTION
**Pull request by tonisojandu**
When only type is defined and no extra matchers then EqualsMatcher was used. This made it impossible to have just a type matcher.

--------

As I have not written Scala before any comments on code style and standard library usage are very welcome.

### CODE REVIEW

File: modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala [hunk 1]
```diff
@@ -90,7 +92,13 @@ object CompareContentsResponseBuilder extends StrictLogging {
           key -> new MatchingRuleGroup(
             rules.rule.flatMap { matchingRule =>
               matchingRule.values.map { struct =>
-                MatchingRule.fromJson(PactCoreUtils.structToJson(toJavaProto(struct)))
+                var json = PactCoreUtils.structToJson(toJavaProto(struct))
```
Comment (austek) [inline modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala]: Could you refactor to not use mutable, please

Comment (tonisojandu) [reply modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala]: Sorry for long response! Vacation period and other duties ate up my attention.

I tried to restructure the code to be more immutable. 

It also turns out that without my change the provide side was not actually doing the matching before, as the `PactCoreUtils.structToJson(toJavaProto(struct))` was producing empty Json-s for everything. After my change I found that it was trying to match using the Avro Utf8 class which was foreign for the Pact matcher which cause provider tests to fail. 

I added a fix for this as well.

File: modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala [hunk 2]
```diff
@@ -90,7 +91,16 @@ object CompareContentsResponseBuilder extends StrictLogging {
...
+                val json = PactCoreUtils.structToJson(toJavaProto(struct))
+                if (json.size() > 0 || matchingRule.`type`.isEmpty) {
+                  MatchingRule.fromJson(json)
+                } else {
+                  MatchingRule.fromJson(
+                    Json.toJson(
+                      Map("match" -> matchingRule.`type`).asJava
+                    )
+                  )
+                }
```
Comment (austek) [inline modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala]: I was thinking sth like
```suggestion
                                MatchingRule.fromJson(
                  Option(PactCoreUtils.structToJson(toJavaProto(struct)))
                    .filterNot(ruleJson => ruleJson.size() == 0 && matchingRule.`type`.nonEmpty) match {
                    case Some(value) => value
                    case None => Json.toJson(Map("match" -> matchingRule.`type`).asJava)
                  }
                )
```

Comment (tonisojandu) [reply modules/plugin/src/main/scala/com/github/austek/plugin/avro/compare/CompareContentsResponseBuilder.scala]: I applied the suggested changes and readjusted the formatting to allow it to pass the`scalafmtCheckAll´.

**Review by austek — APPROVED** [review]

**Pull request merged**